### ISSUE TITLE
rake yard aborts with "singleton can't be dumped"

### ISSUE THREAD

**Issue by janv** [dkubb/axiom#10]
Opening this ticket in response to http://twitter.com/#!/dkubb/status/218702893480550401

What the title says.
I experienced this yesterday after a fresh checkout of veritas:

```
1.9.3-p125-perf ~/development/veritas (master) $ bundle exec rake yard --trace
** Invoke yard (first_time)
** Execute yard
[error]: Missing 'redcarpet' gem for Markdown formatting. Install it with `gem install redcarpet`
[warn]: Syntax error in `license`:(1,18): syntax error, unexpected tinteger, expecting $end
[warn]: ParserSyntaxError: syntax error in `LICENSE`:(1,18): syntax error, unexpected tINTEGER, expecting $end
[warn]: Stack trace:
    /Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/gems/1.9.1/gems/yard-0.8.2.1/lib/yard/parser/ruby/ruby_parser.rb:533:in `on_parse_error'
    /Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/gems/1.9.1/gems/yard-0.8.2.1/lib/yard/parser/ruby/ruby_parser.rb:50:in `parse'
    /Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/gems/1.9.1/gems/yard-0.8.2.1/lib/yard/parser/ruby/ruby_parser.rb:50:in `parse'
    /Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/gems/1.9.1/gems/yard-0.8.2.1/lib/yard/parser/ruby/ruby_parser.rb:15:in `parse'
    /Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/gems/1.9.1/gems/yard-0.8.2.1/lib/yard/parser/source_parser.rb:440:in `parse'
    /Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/gems/1.9.1/gems/yard-0.8.2.1/lib/yard/parser/source_parser.rb:44:in `block in parse'


... lots of warnings omitted


    in file `lib/veritas/relation/operation/insertion.rb' near line 18
[warn]: in YARD::Handlers::Ruby::MixinHandler: Undocumentable mixin: YARD::Parser::UndocumentableError for class Veritas::Relation::Operation::Order::Direction
[warn]:     in file 'lib/veritas/relation/operation/order/direction.rb':11:

    11: include Equalizer.new(:attribute)

[warn]: in YARD::Handlers::Ruby::MixinHandler: Undocumentable mixin: YARD::Parser::UndocumentableError for class Veritas::Relation::Operation::Order::DirectionSet
[warn]:     in file 'lib/veritas/relation/operation/order/direction_set.rb':10:

    10: include Equalizer.new(:to_ary)

rake aborted!
singleton can't be dumped
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/gems/1.9.1/gems/yard-0.8.2.1/lib/yard/serializers/yardoc_serializer.rb:88:in `dump'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/gems/1.9.1/gems/yard-0.8.2.1/lib/yard/serializers/yardoc_serializer.rb:88:in `dump'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/gems/1.9.1/gems/yard-0.8.2.1/lib/yard/serializers/yardoc_serializer.rb:67:in `serialize'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/gems/1.9.1/gems/yard-0.8.2.1/lib/yard/registry_store.rb:173:in `save'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/gems/1.9.1/gems/yard-0.8.2.1/lib/yard/registry.rb:159:in `save'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/gems/1.9.1/gems/yard-0.8.2.1/lib/yard/cli/yardoc.rb:256:in `run'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/gems/1.9.1/gems/yard-0.8.2.1/lib/yard/rake/yardoc_task.rb:69:in `block in define'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/1.9.1/rake/task.rb:205:in `call'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/1.9.1/rake/task.rb:205:in `block in execute'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/1.9.1/rake/task.rb:200:in `each'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/1.9.1/rake/task.rb:200:in `execute'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/1.9.1/rake/task.rb:158:in `block in invoke_with_call_chain'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/1.9.1/monitor.rb:211:in `mon_synchronize'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/1.9.1/rake/task.rb:151:in `invoke_with_call_chain'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/1.9.1/rake/task.rb:144:in `invoke'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/1.9.1/rake/application.rb:116:in `invoke_task'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/1.9.1/rake/application.rb:94:in `block (2 levels) in top_level'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/1.9.1/rake/application.rb:94:in `each'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/1.9.1/rake/application.rb:94:in `block in top_level'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/1.9.1/rake/application.rb:133:in `standard_exception_handling'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/1.9.1/rake/application.rb:88:in `top_level'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/1.9.1/rake/application.rb:66:in `block in run'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/1.9.1/rake/application.rb:133:in `standard_exception_handling'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/1.9.1/rake/application.rb:63:in `run'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/lib/ruby/gems/1.9.1/gems/rake-0.9.2.2/bin/rake:32:in `<top (required)>'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/bin/rake:19:in `load'
/Users/jan/.rbenv/versions/1.9.3-p125-perf/bin/rake:19:in `<main>'
Tasks: TOP => yard
```

Ruby version:

```
1.9.3-p125-perf ~/development/veritas (master) $ ruby --version
ruby 1.9.3p125 (2012-02-16 revision 34643) [x86_64-darwin11.3.0]
```

Thats 1.9.3-p125 with the Falcon patches

Gems:

```
1.9.3-p125-perf ~/development/veritas (master) $ bundle
Using rake (0.9.2.2) 
Using ZenTest (4.8.1) 
Using RubyInline (3.11.2) 
Using sexp_processor (3.0.10) 
Using ParseTree (3.0.8) 
Using backports (2.6.1) 
Using env (0.3.0) 
Using ffi (1.0.11) 
Using ffi-hunspell (0.2.5) 
Using ruby_parser (2.0.6) 
Using flay (1.4.3) 
Using flog (2.5.3) 
Using git (1.2.5) 
Using rb-fchange (0.0.5) 
Using rb-fsevent (0.9.1) 
Using rb-inotify (0.8.8) 
Using listen (0.4.7) 
Using thor (0.15.3) 
Using guard (1.1.1) 
Using bundler (1.1.3) 
Using guard-bundler (0.1.3) 
Using guard-rspec (0.7.3) 
Using json (1.7.3) 
Using rdoc (3.12) 
Using jeweler (1.8.4) 
Using rbench (0.2.3) 
Using ruby2ruby (1.2.2) 
Using reek (1.2.8) from git://github.com/dkubb/reek.git (at master) 
Using roodi (2.1.0) 
Using rspec (1.3.2) 
Using yard (0.8.2.1) 
Using yard-spellcheck (0.1.5) 
Using yardstick (0.5.0) 
Your bundle is complete! Use `bundle show [gemname]` to see where a bundled gem is installed.
```

Rubygems Version is 1.8.11

**Comment by dkubb** [comment]
I've reported this to the YARD project: https://github.com/lsegal/yard/issues/565

**Comment by dkubb** [comment]
@janv in the meantime, I regenerated the YARD docs at: http://rubydoc.info/github/dkubb/veritas/master/frames

**Comment by dkubb** [comment]
I believe the cause of this is some code in Yardstick. I'm going to modify Yardstick and see if I can eliminate this.

**Comment by dkubb** [comment]
Ok, this should work fine with edge Yardstick and Ruby 1.9. The YARD parser for 1.8 still has problems, but it's mostly with statements like `include Equalizer.new(:header, :to_set)`.

I'm going to release Yardstick 0.0.8, bump the dependency to reference that. Since the YARD legacy parser is the cause of the remaining problem once this is done I will close this ticket and make sure it's reported upstream.

**Comment by dkubb** [comment]
@janv ok, I've merged this into master and it fixes the problem under Ruby 1.9. Due to limitations in YARD's legacy 1.8. parser (which I think are less likely to be resolved upstream) there isn't a way to build the docs under 1.8 at the moment.

I'm going to mark this as closed since I've provided the details upstream.

**Comment by janv** [comment]
Cool, Thanks
Had completely forgotten about this :)
